### PR TITLE
dash/sidebar-outside-click

### PR DESCRIPTION
**Pull request by sebastianbochan**
Added closing sidebar, when clicking outside the box.

### CODE REVIEW

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
A few things:
- The mask is missing when the 
- Test is missing 
- You are editing the `BaseFormat` so someone from the Stock team should also review it.

**Comment by sebastianbochan** [comment]
I decided to move the clicking functionality from the main class to the sidebar in dashboards, it was too risky to change annotations popups, etc.

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Works well but the grey overlay `highcharts-dashboards-edit-overlay` is missing.

File: test/cypress/integration/Dashboards/component-add.cy.js [hunk 1]
```diff
@@ -19,6 +19,18 @@ describe('Add components through UI', () => {
         cy.toggleEditMode();
     });
 
+    it('should close the add component sidebar, clicking outside', function() {
+        cy.get('.highcharts-dashboards-edit-tools-btn').contains('Add').click();
+        // cy.get('#dashboard-col-1').first().trigger('click', {force: true});
```
Comment (karolkolodziej) [inline test/cypress/integration/Dashboards/component-add.cy.js:24]: Please remove

File: test/cypress/integration/Dashboards/component-add.cy.js [hunk 2]
```diff
@@ -19,6 +19,18 @@ describe('Add components through UI', () => {
...
+        cy.board().then((board) => {
+            cy.get('.highcharts-dashboards-edit-sidebar').should('exist');
+
```
Comment (karolkolodziej) [inline test/cypress/integration/Dashboards/component-add.cy.js:27]: Please check if the overlay is visible at that point.

**Review by karolkolodziej — APPROVED** [review]

File: test/cypress/integration/Dashboards/component-add.cy.js (hunk 1 above)
Comment (sebastianbochan) [reply test/cypress/integration/Dashboards/component-add.cy.js:24]: Nice catch!

**Review by Vazonik — APPROVED** [review]

**Pull request merged**